src/index.js

```diff
@@ -481,16 +488,44 @@ class EventsCoreAPI {
   }
 
   /**
-   * Authenticating events by verifying signature
+   * Authenticating events by verifying hmac signature
    *
    * @param {object} event JSON payload delivered to the registered webhook URL
    * @param {string} clientSecret Client secret can be retrieved from the Adobe I/O Console integration
-   * @param {string} signatureHeaderValue Value of x-adobe-signature header in each POST request to the registered webhook URL
+   * @param {string} deprecatedSignature Value of x-adobe-signature header in each POST request to the registered webhook URL
+   * @returns {boolean} If signature matches return true else return false
+   */
+  verifySignatureForEvent(event, clientSecret, deprecatedSignature) { 
+    if (clientSecret !== null && typeof (clientSecret) !== 'undefined') {
+      const hmacDigest = Base64.stringify(hmacSHA256(JSON.stringify(event), clientSecret))
+      return hmacDigest === deprecatedSignature
+    }
+  }
+  
+  /**
+   * Authenticating events by verifying digital signature
+   * 
+   * @param {*} event JSON payload delivered to the registered webhook URL
+   * @param {*} recipientClientId Target recipient client id retrieved from the Adobe I/O Console integration
+   * @param {*} signatureOptions map of all digital signature header values consisting fields as below
+   * digiSignature1 : Value of digital signature retrieved from the x-adobe-digital-signature1 header in each POST request to webhook
+   * digiSignature2 : Value of digital signature retrieved from the x-adobe-digital-signature2 header in each POST request to webhook
+   * publicKeyUrl1 : Value of public key url retrieved from the x-adobe-public-key1-url header in each POST request to webhook
+   * publicKeyUrl2 : Value of public key url retrieved from the x-adobe-public-key2-url header in each POST request to webhook
    * @returns {boolean} If signature matches return true else return false
    */
-  verifySignatureForEvent (event, clientSecret, signatureHeaderValue) {
-    const hmacDigest = Base64.stringify(hmacSHA256(JSON.stringify(event), clientSecret))
-    return hmacDigest === signatureHeaderValue
+  async verifyDigitalSignatureForEvent (event, recipientClientId, signatureOptions) {
+    // check event payload and get proper payload used in I/O Events signing
+    const decodedJsonPayload = getProperPayload(event)
+
+    // check if the target recipient is present in event and is a valid one, then verify the signature else return error
+    if (isTargetRecipient(decodedJsonPayload, recipientClientId)) {
+      return await verifyDigitalSignature(signatureOptions.digiSignature1, signatureOptions.digiSignature2,
+        signatureOptions.publicKeyUrl1, signatureOptions.publicKeyUrl2, recipientClientId, JSON.stringify(decodedJsonPayload))
```

Comment:
Why didn't we pass the `signatureOptions` map to `verifyDigitalSignature()` as is?

Reply:
done